### PR TITLE
Move parsing and name validation to cdi/parse package

### PR DESCRIPTION
**Pull request by elezar**
This should allow golang dependencies to be better managed.

See https://github.com/docker/cli/pull/4084#pullrequestreview-1351095415

### CODE REVIEW

File: pkg/cdi/qualified-device.go [hunk 1]
```diff
@@ -17,189 +17,101 @@
...
+	return parser.IsQualifiedName(device)
 }
 
 // ParseQualifiedName splits a qualified name into device vendor, class,
 // and name. If the device fails to parse as a qualified name, or if any
 // of the split components fail to pass syntax validation, vendor and
 // class are returned as empty, together with the verbatim input as the
 // name and an error describing the reason for failure.
+//
+// Deprecated: use parser.Interface.ParseQualifiedName instead
```
Comment (klihub) [inline pkg/cdi/qualified-device.go:58]: Shouldn't this be use `parser.ParseQualifiedName` without `.Interface.` ?

File: pkg/cdi/qualified-device.go [hunk 2]
```diff
@@ -17,189 +17,101 @@
...
-	return vendor, class, name, nil
+	return parser.ParseQualifiedName(device)
 }
 
 // ParseDevice tries to split a device name into vendor, class, and name.
 // If this fails, for instance in the case of unqualified device names,
 // ParseDevice returns an empty vendor and class together with name set
 // to the verbatim input.
+//
+// Deprecated: use parser.Interface.ParseDevice instead
```
Comment (klihub) [inline pkg/cdi/qualified-device.go:68]: ditto here

File: pkg/cdi/qualified-device.go [hunk 3]
```diff
@@ -17,189 +17,101 @@
...
 // ParseQualifier splits a device qualifier into vendor and class.
 // The syntax for a device qualifier is
-//     "<vendor>/<class>"
+//
+//	"<vendor>/<class>"
+//
 // If parsing fails, an empty vendor and the class set to the
 // verbatim input is returned.
+//
+// Deprecated: use parser.Interface.ParseQualifier instead
```
Comment (klihub) [inline pkg/cdi/qualified-device.go:81]: And here.

**Review by klihub — CHANGES_REQUESTED** [review]
I have 3 small nits related to doc strings. Otherwise LGTM.

**Comment by elezar** [comment]
> I have 3 small nits related to doc strings. Otherwise LGTM.

Thanks. Missed updating the comments in my various iterations.

File: pkg/cdi/qualified-device.go (hunk 1 above)
Comment (elezar) [reply pkg/cdi/qualified-device.go:58]: Updated.

**Comment by elezar** [comment]
@klihub thanks for the review. Updated.

**Review by klihub — APPROVED** [review]

**Pull request merged**